### PR TITLE
🐛(frontend) Bug/error message course run item

### PR DESCRIPTION
**Pull request by PanchoutNathan**
## Purpose

On the CourseProductItem, when user tries to enroll or unenroll to a course run, if the request fails, the user has no feedback the action is failing that is weird.

## Proposal

An error message should be display to explicitly show to the user that its action has failed.

### CODE REVIEW

File: src/frontend/js/components/CourseProductCourseRuns/EnrollableCourseRunList.tsx [hunk 1]
```diff
@@ -61,7 +66,9 @@ const EnrollableCourseRunList = ({ courseRuns, order }: Props) => {
   const formRef = useRef<HTMLFormElement>(null);
 
   const [selectedCourseRun, setSelectedCourseRun] = useState<Maybe<Joanie.CourseRun>>();
+  const [errors, setErrors] = useState<MessageDescriptor[]>([]);
```
Comment (jbpenrath) [inline src/frontend/js/components/CourseProductCourseRuns/EnrollableCourseRunList.tsx:69]: I think we should update the `useEnrollment` hook to base it from `useResource`. In this way, we are able to retrieve an error message on get / create / update / delete enrollment.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Fix courses badges css.
 - Fix style in edit mode on courses with catalog visibility with
   `course_only` or `course_and_search`.
+- Show error message when user tries to enroll or unenroll to a 
+  course run and the requests fails
```
Comment (jbpenrath) [inline CHANGELOG.md:29]: Move on the top of entries

File: CHANGELOG.md [hunk 3]
```diff
@@ -25,6 +25,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Fix courses badges css.
```
Comment (jbpenrath) [inline CHANGELOG.md:25]: In your commit message :
```diff
- On the CourseProductItem, when user tries to enroll or enunroll to a course run,
+ On the CourseProductItem, when user tries to enroll or unroll to a course run,

```

Then add in the commit footer
`Close #1858`

File: src/frontend/js/components/CourseProductCourseRuns/EnrolledCourseRun.tsx [hunk 4]
```diff
@@ -70,6 +70,7 @@ const EnrolledCourseRun = ({ enrollment }: Props) => {
             {formatDate(enrollment.course_run.end)}
           </em>
         </div>
+        {states.error && <div className="course-runs-list__errors-feedback ">{states.error}</div>}
```
Comment (jbpenrath) [inline src/frontend/js/components/CourseProductCourseRuns/EnrolledCourseRun.tsx:73]: A `<p>` makes more sens here.
```suggestion
        {states.error && <p className="course-runs-list__errors-feedback ">{states.error}</p>}
```

Comment (PanchoutNathan) [reply src/frontend/js/components/CourseProductCourseRuns/EnrolledCourseRun.tsx:73]: agreed

File: CHANGELOG.md (hunk 3 above)
Comment (PanchoutNathan) [reply CHANGELOG.md:25]: oops

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**